Accessibility labels for form fields and fieldsets

arnelap opened the pull request:
What I changed and tested > see comment below. 

I propose to wrap every input in its label, that way we don't need to deal with ID's
this way we prevent any issues with duplicated IDs and are still w3c accessible compliant by Associating labels implicitly
https://www.w3.org/WAI/tutorials/forms/labels/#using-aria-label

to this end I
- Removed the ID from the default form content 
- Wrapped inputs in their label
- Added "fieldset" around checkbox and radios with the agree to terms excluded as that has only 1 choice it can still be wrapped in label
- removed the wrap in P option for "fieldset" as a fieldset will close the p / can't be wrapped in p

  this would close #640

arnelap committed:
Associating labels implicitly by wrapping the inputs in labels and applying fieldset to checkboxes/radios

also changed the default form to not include any ID as that would cause invalid HTML if there are multiple forms on one page.

arnelap committed:
it aint pretty(print) but it's the only way i managed to get the /label to wrap to new line for terms agreement

arnelap committed:
my editor removed a return at the end of the file :/

arnelap committed:
es lint stuff

arnelap committed:
fixed the return before /label without a replace

dannyvankooten commented on assets/src/js/admin/form-editor/field-forms.js:
```diff
@@ -47,7 +47,9 @@ forms.choice = function (config) {
+  if (config.type !== 'checkbox') {
+    visibleRows.push(rows.useParagraphs(config))
+  }
 
```
The option "wrap in paragraph tags" is shown for field of type `radio`, but not doing anything. We should probably remove it for type `radio` too.

dannyvankooten commented on assets/src/js/admin/form-editor/field-generator.js:
```diff
@@ -72,7 +72,7 @@ generators['terms-checkbox'] = function (config) {
     }),
     ' ',
-    label
+    m('span', label)
```
Why do we wrap the label in `<span>` here now?

dannyvankooten committed:
Update field-generator.js

arnelap committed:
Wrapping label text in span, and updating default form for the same. Also some CSS changes to target that span instead of label

dannyvankooten commented:
Yes

On 15 December 2025 15:16:19 CET, Arne Lap ***@***.***> wrote:
>@arnelap commented on this pull request.
>
>
>
>> +  const content = config.type === 'terms-checkbox'
>+    ? field
>+    : isNested
>+      ? (hasLabel
>+          ? m('label', [config.label, field])
>
>I believe this is resolved with your push? 
>

arnelap committed:
CSS to support both new and old style

Also made change to your last edit in form-generator to prevent a js error when adding fields and changed the default form content to be consistent.

arnelap commented:
so we will always have the issue of Premium add-on being "out of sync" with this sort of update, I think the best solution we can offer is first publishing the Premium add-on update, wait a bit and then publish this update?

That is the main reason for me to leave this on draft for now, besides it still warranting some more testing I suppose.

dannyvankooten committed:
Merge branch 'main' into accessibility-labels
